### PR TITLE
feat: time based settlements

### PR DESCRIPTION
**Pull request by ralph-pichler**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/1711)

### CODE REVIEW

File: pkg/debugapi/router.go [hunk 1]
```diff
@@ -125,15 +125,18 @@ func (s *Service) newRouter() *mux.Router {
 		"GET": http.HandlerFunc(s.peerBalanceHandler),
 	})
 
-	router.Handle("/settlements", jsonhttp.MethodHandler{
-		"GET": http.HandlerFunc(s.settlementsHandler),
-	})
-
-	router.Handle("/settlements/{peer}", jsonhttp.MethodHandler{
-		"GET": http.HandlerFunc(s.peerSettlementsHandler),
+	router.Handle("/pseudosettlements", jsonhttp.MethodHandler{
```
Comment (Eknir) [inline pkg/debugapi/router.go:128]: Let's give this variable (user-exposed) a more proper name. We can leave the internal names at pseudosettlements.

suggestion: `timesettlements`

File: cmd/bee/cmd/cmd.go [hunk 2]
```diff
@@ -217,7 +217,7 @@ func (c *command) setAllFlags(cmd *cobra.Command) {
 	cmd.Flags().String(optionWelcomeMessage, "", "send a welcome message string during handshakes")
 	cmd.Flags().Bool(optionNameGlobalPinningEnabled, false, "enable global pinning")
 	cmd.Flags().String(optionNamePaymentThreshold, "10000000000000", "threshold in BZZ where you expect to get paid from your peers")
-	cmd.Flags().String(optionNamePaymentTolerance, "50000000000000", "excess debt above payment threshold in BZZ where you disconnect from your peer")
+	cmd.Flags().String(optionNamePaymentTolerance, "10000000000000", "excess debt above payment threshold in BZZ where you disconnect from your peer")
```
Comment (Eknir) [inline cmd/bee/cmd/cmd.go:220]: Please adjust also all variable and variable-comments in our packages configuration files.

File: pkg/accounting/accounting.go [hunk 3]
```diff
@@ -76,10 +99,21 @@ type Accounting struct {
 	// The amount in BZZ we let peers exceed the payment threshold before we
 	// disconnect them.
 	paymentTolerance *big.Int
-	earlyPayment     *big.Int
-	payFunction      PayFunc
-	pricing          pricing.Interface
-	metrics          metrics
+	// Start settleing when reserve plus debt reaches this close to threshold.
```
Comment (Eknir) [inline pkg/accounting/accounting.go:102]: typo

File: pkg/accounting/accounting.go [hunk 4]
```diff
@@ -103,16 +135,21 @@ func NewAccounting(
...
 		paymentTolerance: new(big.Int).Set(PaymentTolerance),
 		earlyPayment:     new(big.Int).Set(EarlyPayment),
+		disconnectLimit:  new(big.Int).Add(PaymentThreshold, PaymentTolerance),
 		logger:           Logger,
 		store:            Store,
 		pricing:          Pricing,
 		metrics:          newMetrics(),
+		refreshRate:      refreshRate,
+		timeNow:          time.Now,
+		minimumPayment:   new(big.Int).Div(refreshRate, big.NewInt(5)),
```
Comment (Eknir) [inline pkg/accounting/accounting.go:152]: Consider setting this as a constant, for readability

File: pkg/accounting/accounting.go [hunk 5]
```diff
@@ -243,103 +278,54 @@ func (a *Accounting) settle(ctx context.Context, peer swarm.Address, balance *ac
...
 		}
-
 	}
 
-	currentBalance, err := a.Balance(peer)
-	if err != nil {
-		if !errors.Is(err, ErrPeerNoBalance) {
-			return fmt.Errorf("failed to load balance: %w", err)
+	if a.payFunction != nil {
+		if !balance.paymentOngoing {
```
Comment (Eknir) [inline pkg/accounting/accounting.go:315]: Can be in line with `314` to reduce nesting.

File: pkg/accounting/accounting.go (hunk 4 above)
Comment (Eknir) [reply pkg/accounting/accounting.go:152]: Please also add the reasoning behind this specific value of the `minimumPayment`

File: cmd/bee/cmd/cmd.go (hunk 2 above)
Comment (ralph-pichler) [reply cmd/bee/cmd/cmd.go:221]: donee

File: pkg/accounting/accounting.go (hunk 3 above)
Comment (ralph-pichler) [reply pkg/accounting/accounting.go]: fixed

File: pkg/accounting/accounting.go (hunk 5 above)
Comment (ralph-pichler) [reply pkg/accounting/accounting.go]: done

File: pkg/debugapi/router.go (hunk 1 above)
Comment (ralph-pichler) [reply pkg/debugapi/router.go]: done

File: pkg/accounting/accounting.go (hunk 4 above)
Comment (ralph-pichler) [reply pkg/accounting/accounting.go]: done

**Review by Eknir — APPROVED** [review]
lgtm. Reviewed together with @anatollupacescu .

**Pull request merged**